htdp-lib/2htdp/abstraction.rkt

```diff
@@ -114,7 +115,33 @@
     [(_ type:id case:expr [variant:id (field:id ...) body:expr] ...)
      #'(type-case type case [variant (field ...) body] ...)]))
 
+(begin-for-syntax
+  (define-syntax-class pattern-cls
+    #:description "pattern"
+    #:attributes (expr)
+    (pattern name:id
+             #:with expr #'name)
+
+    ;; literal-constants
+    (pattern {~or* :string :number :boolean :char :regexp :bytes :byte-regexp}
+             #:with expr this-syntax)
+    (pattern ({~datum quote} x)
+             #:with expr #'(quote x))
+
+    (pattern ({~datum cons} left:pattern-cls right:pattern-cls)
```

Comment:
Do you want to also allow `list` patterns? What forms of them?

Reply:
This is the only outstanding issue. @mfelleisen, do you have any thought on this? The grammar currently doesn't allow the `list` pattern, but I can add it if you prefer.

Reply:
So sorry. This email fell thru the cracks. Yes, let's add the `list` shorthand too. Thanks.

Reply:
I will add that. Should it also support ellipses?

Reply:
NO :-)